web/src/main/java/se/inera/intyg/webcert/web/csintegration/certificate/RenewCertificateFromCertificateService.java

```diff
@@ -0,0 +1,94 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package se.inera.intyg.webcert.web.csintegration.certificate;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import se.inera.intyg.common.support.common.enumerations.HandelsekodEnum;
+import se.inera.intyg.webcert.web.csintegration.integration.CSIntegrationRequestFactory;
+import se.inera.intyg.webcert.web.csintegration.integration.CSIntegrationService;
+import se.inera.intyg.webcert.web.csintegration.util.PDLLogService;
+import se.inera.intyg.webcert.web.service.facade.RenewCertificateFacadeService;
+import se.inera.intyg.webcert.web.service.monitoring.MonitoringLogService;
+import se.inera.intyg.webcert.web.service.user.WebCertUserService;
+import se.inera.intyg.webcert.web.web.controller.integration.dto.IntegrationParameters;
+
+@Slf4j
+@Service("renewCertificateFromCertificateService")
+@RequiredArgsConstructor
+public class RenewCertificateFromCertificateService implements RenewCertificateFacadeService {
+
+    private final CSIntegrationService csIntegrationService;
+    private final CSIntegrationRequestFactory csIntegrationRequestFactory;
+    private final PDLLogService pdlLogService;
+    private final MonitoringLogService monitoringLogService;
+    private final WebCertUserService webCertUserService;
+    private final IntegratedUnitRegistryHelper integratedUnitRegistryHelper;
+    private final PublishCertificateStatusUpdateService publishCertificateStatusUpdateService;
+
+    @Override
+    public String renewCertificate(String certificateId) {
+        log.debug("Attempting to renew certificate '{}' from Certificate Service", certificateId);
+
+        if (Boolean.FALSE.equals(csIntegrationService.certificateExists(certificateId))) {
+            log.debug("Certificate '{}' does not exist in certificate service", certificateId);
+            return null;
+        }
+
+        final var certificateToRenew = csIntegrationService.getCertificate(
+            certificateId,
+            csIntegrationRequestFactory.getCertificateRequest()
+        );
+        if (certificateToRenew == null) {
+            throw new IllegalStateException(
+                String.format("Certificate service returned null when getting certificate '%s'", certificateId)
+            );
+        }
+
+        final var integrationParameters = webCertUserService.getUser().getParameters();
+        final var renewalCertificate = csIntegrationService.renewCertificate(
+            certificateId,
+            csIntegrationRequestFactory.renewCertificateRequest(
+                isAlternateSSNDefined(integrationParameters) ? integrationParameters.getAlternateSsn()
```

Comment:
I found some code in the current facade-implementation that manages a situation if the personId is a "reserve id". See line 71 in RenewCertificateFacadeServiceImpl. Good to look at the whole commit that includes changes in multiple classes.

Reply:
No, I don't think it is necessary. Any reserveIds handled by Webcert due to IntegrationParameters should stay in Webcert.